### PR TITLE
Track 3, ROCm Robotics, Franka Multi-Fruit Sorting via SmolVLA on AMD ROCm

### PR DESCRIPTION
**Pull request by yigenfeng0707-netizen**
## Track 3 Submission — Franka Multi-Fruit Sorting via SmolVLA on AMD ROCm

**Team**: ROCm Robotics (Solo)
**Track**: Track 3 — Physical AI Challenge
**Application**: Franka Multi-Fruit Sorting via SmolVLA on AMD ROCm

## Deliverables

| # | Deliverable | Location | Status |
|---|-------------|----------|--------|
| 1 | Technical Report (8 sections, PDF) | `docs/Technical_Report.pdf` | ✅ |
| 2 | Project Source Code + Dockerfile | `submission/`, `Dockerfile` | ✅ |
| 3 | Reproducibility README (4 parts) | `README.md` | ✅ |
| 4 | Demo Video (4 min, 1080p) | `docs/demo_video.mp4` | ✅ |
| 5 | Supplementary: Upstream Contribution (Issue + PR) | `docs/upstream_contribution_evidence.png` | ✅ |

## Scoring Dimensions Coverage

| Dimension | Points | Evidence |
|-----------|--------|----------|
| Robot Capability Performance | 30 | 100% closed-loop eval success rate (2/2 episodes) — `logs/eval_results.json` + Demo Video |
| AMD Radeon GPU & ROCm Adoption | 20 | Technical Report §4 + `logs/smolvla_train.log` (ROCm 7.2.1, W7900, 5min training) |
| Innovation | 20 | Technical Report §5 (6 key technical contributions) |
| Application Value | 20 | Technical Report §1, §5 (logistics sorting, food processing, Agriculture 4.0) |
| Upstream Open-Source Contributions | 10 | **huggingface/lerobot Issue #4205 + Pull Request #4324** (ROCm uint8 bilinear bug + fix) |

## Key Results

- **Success Rate**: 100% (2/2 episodes)
- **Training**: 2000 steps, final loss 0.073, ~5 minutes on W7900
- **Platform**: AMD ROCm 7.2.1 + PyTorch 2.9.1+rocm7.2.1
- **Tech Stack**: Genesis 1.2.3 + LeRobot 0.6.0 + SmolVLA (all Apache-2.0)

## Upstream Contribution (Issue + Pull Request)

We contributed to `huggingface/lerobot` by **both filing an issue AND opening a fix Pull Request**:

- **Issue #4205** (submitted 2026-07-29, OPEN): https://github.com/huggingface/lerobot/issues/4205
  - Title: [ROCm] NotImplementedError for bilinear interpolate on uint8 tensor in SmolVLA resize_with_pad during manual inference
- **Pull Request #4324** (submitted 2026-08-04, OPEN): https://github.com/huggingface/lerobot/pull/4324
  - Title: Fix uint8 bilinear interpolate NotImplementedError on ROCm
  - Branch: `yigenfeng0707-netizen:fix/rocm-uint8-bilinear-interpolate` → `huggingface:main`
  - Commit SHA: `bfb3487f3b37d64be44dae62075d40247779b08b`
  - Closes #4205
- **Direction**: Improve AMD ROCm platform support (PyTorch ROCm backend missing uint8 bilinear interpolation kernel)
- **Fix artifacts**: `docs/upstream_fix.patch`, `docs/upstream_pr_description.md`
- **Local runtime shim**: `submission/src/utils/rocm_resize_patch.py` (monkey-patches `resize_with_pad` to cast uint8 to float32 before `F.interpolate`, used by the enhanced evaluator `eval_sort_smolvla_robust.py`)

## Reproduction

See `README.md` for full step-by-step reproduction guide. Quick start:

```bash
# 1. Provision Radeon Cloud W7900 instance with ROCm 7.2.1
# 2. Clone this repo + starter repo
# 3. Install dependencies: pip install -r submission/requirements.txt
# 4. Run: python submission/src/scene/sort_demo.py --save-frames
# 5. Collect: python submission/src/data/record_sort_dataset.py --episodes 5 --fps 30
# 6. Train: python submission/src/train/run_smolvla_train.py
# 7. Eval: python submission/src/eval/eval_sort_smolvla.py --checkpoint 002000
```

All submission materials are under the `track3_ROCM_Robotics_FrankaFruitSort/` directory.

### CODE REVIEW

**Comment by yigenfeng0707-netizen** [comment]
Closing in favor of PR #198 which resolves README conflict
